src/index.js

```diff
@@ -30,50 +30,36 @@ export default function build (babel: Object): Object {
 
   return new Transformer("typecheck", {
     Program (node: Object, parent: Object, scope: Object) {
-      try {
-        this.traverse(visitors, {
-          constants: scope.getAllBindingsOfKind("const"),
-          subject: node,
-        });
-      }
-      catch (e) {
-        if (e instanceof SyntaxError) {
-          throw this.errorWithNode(e.message);
-        }
-        else {
-          throw e;
-        }
-      }
+      this.traverse(visitors, {
+        constants: scope.getAllBindingsOfKind("const"),
+        subject: node,
+      });
     },
     Function (node: Object, parent: Object, scope: Object) {
-      try {
-        if (node.typeParameters != null) {
-          throw this.errorWithNode('Type parameters are not supported');
-        }
+      if (node.typeParameters != null) {
```

Comment:
This also applies to https://github.com/codemix/babel-plugin-typecheck/pull/26/files#diff-1fdf421c05c1140f6d71444ea2b27638R654

If we can't insert a typecheck there, we should just ignore it. We could potentially add a `verbose` option which logged this to stderr.